src/libcore/intrinsics.rs

```diff
@@ -545,3 +565,32 @@ extern "rust-intrinsic" {
     /// Performs checked `u64` multiplication.
     pub fn u64_mul_with_overflow(x: u64, y: u64) -> (u64, bool);
 }
+
+/// Inform the optimizer that `expected_val` is expected
+/// (the most probable) value of val.
+#[cfg(stage0)]
+pub unsafe fn expect8(val: u8, _expected_val: u8) -> u8 {
```

Comment:
These functions aren't called during the bootstrap, so you can have the FFI declarations be `#[cfg(not(stage0))]` and just omit these altogether. (I.e. have `core::intrinsics::expect*` not exist in `stage0` at all.)